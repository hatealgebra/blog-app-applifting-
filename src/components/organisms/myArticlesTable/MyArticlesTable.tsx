import React from "react";

import { useAppDispatch, useAppSelector } from "../../../store/hooks";
// import { selectMyArticlesItems } from "../../../store/admin/myArticles.slice";
import { getArticlesFeedThunk } from "../../../store/thunks/articles.thunk";
import { ADMIN_LINKS } from "../../../utils/contants";
import AdminHeading from "../../molecules/adminHeading/AdminHeading";
import EditArticleRow from "../../molecules/editArticleRow/EditArticleRow";
import EditArticleRowButtons from "../../molecules/editArticleRow/EditArticleRowButtons";
import {
  MyArticlesForm,
  MyArticlesTableContainer,
  StyledArticlesTable,
  StyledFallbackContentContainer,
} from "./myArticlesTable.styled";
import {
  selectMyArticlesItems,
  selectMyArticlesOriginalItems,
  selectMyArticlesStatus,
  setArticleToEdit,
} from "../../../store/slices/admin.slices";
import { components } from "../../../types/declarations";

import { deleteArticleThunk } from "../../../store/thunks/admin.thunks";

import noArticles from "../../../images/no-articles.png";
import Loading from "../../atoms/loadingIcon/Loading";
import { navigate } from "gatsby";
import { selectAuthToken } from "../../../store/slices/auth.slices";

// TODO: Finish the my Articles
// TODO: Multiple delete action?
const MyArticlesTable = () => {
  const [checkedBoxes, setCheckedBoxes] = React.useState<boolean[]>([]);
  const originalArray = useAppSelector(selectMyArticlesOriginalItems);
  const articles = useAppSelector(selectMyArticlesItems);
  const status = useAppSelector(selectMyArticlesStatus);
  const dispatch = useAppDispatch();
  const access_token = useAppSelector(selectAuthToken);

  const deleteArticle = (articleId: string) =>
    dispatch(deleteArticleThunk({ articleId, originalArray, access_token }));
  const editArticle = (article: components["schemas"]["Article"]) => {
    dispatch(setArticleToEdit(article));
    navigate(ADMIN_LINKS.EDIT_ARTICLE);
  };

  const switchAllBoxes = (isChecked: boolean) => {
    const allBoxesChecked = checkedBoxes.map((value) => true);
    const allBoxesUncheck = checkedBoxes.map((value) => false);
    if (isChecked) {
      return setCheckedBoxes(allBoxesChecked);
    }
    return setCheckedBoxes(allBoxesUncheck);
  };
  React.useEffect(() => {
    !originalArray && dispatch(getArticlesFeedThunk());
  }, [articles]);

  return (
    <MyArticlesTableContainer>
      <AdminHeading
        heading="My Articles"
        buttonText="Create new article"
        to={ADMIN_LINKS.CREATE_ARTICLE}
      />
      <MyArticlesForm onSubmit={(e) => e.preventDefault()}>
        <StyledArticlesTable>
          <EditArticleRowButtons
            originalArray={originalArray}
            switchAllBoxes={switchAllBoxes}
            dispatch={dispatch}
          />
          {status === "loading" ? (
            <StyledFallbackContentContainer>
              <Loading />
            </StyledFallbackContentContainer>
          ) : articles !== undefined && articles.length > 0 ? (
            articles.map(
              (article: components["schemas"]["ArticleDetail"], i) => {
                const { articleId, title, perex, comments } = article;
                return (
                  <EditArticleRow
                    key={articleId}
                    iteration={i}
                    articleId={articleId}
                    title={title}
                    perex={perex}
                    comments={comments!.length}
                    deleteArticle={deleteArticle}
                    editArticle={() => editArticle(article)}
                    isChecked={checkedBoxes[i]}
                    setCheckedBoxes={setCheckedBoxes}
                  />
                );
              }
            )
          ) : articles && articles.length === 0 ? (
            <StyledFallbackContentContainer>
              <img src={noArticles} alt="No articles, you should cooksome" />
              <span>No articles written yet, you should cook some!</span>
            </StyledFallbackContentContainer>
          ) : (
            <StyledFallbackContentContainer>
              Error
            </StyledFallbackContentContainer>
          )}
        </StyledArticlesTable>
      </MyArticlesForm>
    </MyArticlesTableContainer>
  );
};

export default MyArticlesTable;
