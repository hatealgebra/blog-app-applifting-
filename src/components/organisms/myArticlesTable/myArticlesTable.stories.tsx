import React from 'react';
import { userEvent, within } from '@storybook/testing-library';
import MyArticlesTable from './MyArticlesTable';

// TODO: Add comments play story
// TODO: Add specific msw mocking for these stories, to show different

const Template = () => <MyArticlesTable />;

export const MyArticlesTableExample = Template.bind({});
export const MyArticleTableLoading = Template.bind({});

// export const NoArticles = () => <MyArticlesTable />;
// NoArticles.parameters = {
//   msw: {
//     handlers: [
//       rest.get(`${BASE_API_URL}/articles`, (req, res, ctx) => {
//         return res(ctx.status(200), ctx.json([]));
//       }),
//     ],
//   },
// };

export const SortByTitle = Template.bind({});
SortByTitle.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement);
  const sortBtnTitle = canvas.getByRole('button', { name: 'Article title' });
  await userEvent.click(sortBtnTitle);
};

export const SortByPerex = Template.bind({});
SortByPerex.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement);
  const SortBtnPerex = canvas.getByRole('button', { name: 'Perex' });
  await userEvent.click(SortBtnPerex);
};

export default {
  title: 'Organisms/My Articles Table',
  component: MyArticlesTable,
};
