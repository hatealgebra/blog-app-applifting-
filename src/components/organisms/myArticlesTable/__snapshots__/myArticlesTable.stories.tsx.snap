// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/My Articles MyArticleTableLoading smoke-test 1`] = `
<div class="sc-jdkVqZ jsYxwb">
  <div class="sc-csmVar huQTvo">
    <h1>
      My Articles
    </h1>
    <a colortheme="primary"
       class="sc-hLclGa sc-pqitP dhTwVn bNVlce"
       href="/admin/create-article"
    >
      Create new article
    </a>
  </div>
  <form class="sc-ciJnBw dfwyrO">
    <table class="sc-cCIIiH jupHLD">
      <tr class="sc-hHvkSs OHqgu edit-article">
        <th class="edit-article__checkbox">
          <div class="sc-jSwlEQ hPvNpH">
            <input type="checkbox"
                   class="sc-eDnVMP jVxqso"
            >
          </div>
        </th>
        <th class="edit-article__title">
          <button class="sc-grXZZQ kImwcT">
            <span>
              Article title
            </span>
            <svg stroke="currentColor"
                 fill="currentColor"
                 stroke-width="0"
                 viewbox="0 0 320 512"
                 height="1em"
                 width="1em"
                 xmlns="http://www.w3.org/2000/svg"
            >
              <path d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z">
              </path>
            </svg>
          </button>
        </th>
        <th class="edit-article__perex">
          <button class="sc-grXZZQ kImwcT">
            <span>
              Perex
            </span>
            <svg stroke="currentColor"
                 fill="currentColor"
                 stroke-width="0"
                 viewbox="0 0 320 512"
                 height="1em"
                 width="1em"
                 xmlns="http://www.w3.org/2000/svg"
            >
              <path d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z">
              </path>
            </svg>
          </button>
        </th>
        <th class="edit-article__author">
          <button class="sc-grXZZQ kImwcT">
            <span>
              Author
            </span>
            <svg stroke="currentColor"
                 fill="currentColor"
                 stroke-width="0"
                 viewbox="0 0 320 512"
                 height="1em"
                 width="1em"
                 xmlns="http://www.w3.org/2000/svg"
            >
              <path d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z">
              </path>
            </svg>
          </button>
        </th>
        <th class="edit-article__nr-comments">
          <button class="sc-grXZZQ kImwcT">
            <span>
              # comments
            </span>
            <svg stroke="currentColor"
                 fill="currentColor"
                 stroke-width="0"
                 viewbox="0 0 320 512"
                 height="1em"
                 width="1em"
                 xmlns="http://www.w3.org/2000/svg"
            >
              <path d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z">
              </path>
            </svg>
          </button>
        </th>
        <th class="edit-article__actions-container">
          Actions
        </th>
      </tr>
      // TODO: Detach logic of rendering
      <div class="sc-bYUneI sc-gXCJSa dWksIS cuYcqn">
        <div data-testid="three-dots-loading"
             aria-label="three-dots-loading"
             aria-busy="true"
             role="status"
             style="display: flex;"
        >
          <svg width="80"
               height="80"
               viewbox="0 0 120 30"
               xmlns="http://www.w3.org/2000/svg"
               fill="#007BFF"
               data-testid="three-dots-svg"
          >
            <circle cx="15"
                    cy="15"
                    r="15"
            >
              <animate attributename="r"
                       from="15"
                       to="15"
                       begin="0s"
                       dur="0.8s"
                       values="15;9;15"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
              <animate attributename="fill-opacity"
                       from="1"
                       to="1"
                       begin="0s"
                       dur="0.8s"
                       values="1;.5;1"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
            </circle>
            <circle cx="60"
                    cy="15"
                    r="9"
                    attributename="fill-opacity"
                    from="1"
                    to="0.3"
            >
              <animate attributename="r"
                       from="9"
                       to="9"
                       begin="0s"
                       dur="0.8s"
                       values="9;15;9"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
              <animate attributename="fill-opacity"
                       from="0.5"
                       to="0.5"
                       begin="0s"
                       dur="0.8s"
                       values=".5;1;.5"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
            </circle>
            <circle cx="105"
                    cy="15"
                    r="15"
            >
              <animate attributename="r"
                       from="15"
                       to="15"
                       begin="0s"
                       dur="0.8s"
                       values="15;9;15"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
              <animate attributename="fill-opacity"
                       from="1"
                       to="1"
                       begin="0s"
                       dur="0.8s"
                       values="1;.5;1"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
            </circle>
          </svg>
        </div>
      </div>
    </table>
  </form>
</div>
`;

exports[`Organisms/My Articles MyArticlesTableExample smoke-test 1`] = `
<div class="sc-jdkVqZ jsYxwb">
  <div class="sc-csmVar huQTvo">
    <h1>
      My Articles
    </h1>
    <a colortheme="primary"
       class="sc-hLclGa sc-pqitP dhTwVn bNVlce"
       href="/admin/create-article"
    >
      Create new article
    </a>
  </div>
  <form class="sc-ciJnBw dfwyrO">
    <table class="sc-cCIIiH jupHLD">
      <tr class="sc-hHvkSs OHqgu edit-article">
        <th class="edit-article__checkbox">
          <div class="sc-jSwlEQ hPvNpH">
            <input type="checkbox"
                   class="sc-eDnVMP jVxqso"
            >
          </div>
        </th>
        <th class="edit-article__title">
          <button class="sc-grXZZQ kImwcT">
            <span>
              Article title
            </span>
            <svg stroke="currentColor"
                 fill="currentColor"
                 stroke-width="0"
                 viewbox="0 0 320 512"
                 height="1em"
                 width="1em"
                 xmlns="http://www.w3.org/2000/svg"
            >
              <path d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z">
              </path>
            </svg>
          </button>
        </th>
        <th class="edit-article__perex">
          <button class="sc-grXZZQ kImwcT">
            <span>
              Perex
            </span>
            <svg stroke="currentColor"
                 fill="currentColor"
                 stroke-width="0"
                 viewbox="0 0 320 512"
                 height="1em"
                 width="1em"
                 xmlns="http://www.w3.org/2000/svg"
            >
              <path d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z">
              </path>
            </svg>
          </button>
        </th>
        <th class="edit-article__author">
          <button class="sc-grXZZQ kImwcT">
            <span>
              Author
            </span>
            <svg stroke="currentColor"
                 fill="currentColor"
                 stroke-width="0"
                 viewbox="0 0 320 512"
                 height="1em"
                 width="1em"
                 xmlns="http://www.w3.org/2000/svg"
            >
              <path d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z">
              </path>
            </svg>
          </button>
        </th>
        <th class="edit-article__nr-comments">
          <button class="sc-grXZZQ kImwcT">
            <span>
              # comments
            </span>
            <svg stroke="currentColor"
                 fill="currentColor"
                 stroke-width="0"
                 viewbox="0 0 320 512"
                 height="1em"
                 width="1em"
                 xmlns="http://www.w3.org/2000/svg"
            >
              <path d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z">
              </path>
            </svg>
          </button>
        </th>
        <th class="edit-article__actions-container">
          Actions
        </th>
      </tr>
      // TODO: Detach logic of rendering
      <div class="sc-bYUneI sc-gXCJSa dWksIS cuYcqn">
        <div data-testid="three-dots-loading"
             aria-label="three-dots-loading"
             aria-busy="true"
             role="status"
             style="display: flex;"
        >
          <svg width="80"
               height="80"
               viewbox="0 0 120 30"
               xmlns="http://www.w3.org/2000/svg"
               fill="#007BFF"
               data-testid="three-dots-svg"
          >
            <circle cx="15"
                    cy="15"
                    r="15"
            >
              <animate attributename="r"
                       from="15"
                       to="15"
                       begin="0s"
                       dur="0.8s"
                       values="15;9;15"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
              <animate attributename="fill-opacity"
                       from="1"
                       to="1"
                       begin="0s"
                       dur="0.8s"
                       values="1;.5;1"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
            </circle>
            <circle cx="60"
                    cy="15"
                    r="9"
                    attributename="fill-opacity"
                    from="1"
                    to="0.3"
            >
              <animate attributename="r"
                       from="9"
                       to="9"
                       begin="0s"
                       dur="0.8s"
                       values="9;15;9"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
              <animate attributename="fill-opacity"
                       from="0.5"
                       to="0.5"
                       begin="0s"
                       dur="0.8s"
                       values=".5;1;.5"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
            </circle>
            <circle cx="105"
                    cy="15"
                    r="15"
            >
              <animate attributename="r"
                       from="15"
                       to="15"
                       begin="0s"
                       dur="0.8s"
                       values="15;9;15"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
              <animate attributename="fill-opacity"
                       from="1"
                       to="1"
                       begin="0s"
                       dur="0.8s"
                       values="1;.5;1"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
            </circle>
          </svg>
        </div>
      </div>
    </table>
  </form>
</div>
`;

exports[`Organisms/My Articles NoArticles smoke-test 1`] = `
<div class="sc-jdkVqZ jsYxwb">
  <div class="sc-csmVar huQTvo">
    <h1>
      My Articles
    </h1>
    <a colortheme="primary"
       class="sc-hLclGa sc-pqitP dhTwVn bNVlce"
       href="/admin/create-article"
    >
      Create new article
    </a>
  </div>
  <form class="sc-ciJnBw dfwyrO">
    <table class="sc-cCIIiH jupHLD">
      <tr class="sc-hHvkSs OHqgu edit-article">
        <th class="edit-article__checkbox">
          <div class="sc-jSwlEQ hPvNpH">
            <input type="checkbox"
                   class="sc-eDnVMP jVxqso"
            >
          </div>
        </th>
        <th class="edit-article__title">
          <button class="sc-grXZZQ kImwcT">
            <span>
              Article title
            </span>
            <svg stroke="currentColor"
                 fill="currentColor"
                 stroke-width="0"
                 viewbox="0 0 320 512"
                 height="1em"
                 width="1em"
                 xmlns="http://www.w3.org/2000/svg"
            >
              <path d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z">
              </path>
            </svg>
          </button>
        </th>
        <th class="edit-article__perex">
          <button class="sc-grXZZQ kImwcT">
            <span>
              Perex
            </span>
            <svg stroke="currentColor"
                 fill="currentColor"
                 stroke-width="0"
                 viewbox="0 0 320 512"
                 height="1em"
                 width="1em"
                 xmlns="http://www.w3.org/2000/svg"
            >
              <path d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z">
              </path>
            </svg>
          </button>
        </th>
        <th class="edit-article__author">
          <button class="sc-grXZZQ kImwcT">
            <span>
              Author
            </span>
            <svg stroke="currentColor"
                 fill="currentColor"
                 stroke-width="0"
                 viewbox="0 0 320 512"
                 height="1em"
                 width="1em"
                 xmlns="http://www.w3.org/2000/svg"
            >
              <path d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z">
              </path>
            </svg>
          </button>
        </th>
        <th class="edit-article__nr-comments">
          <button class="sc-grXZZQ kImwcT">
            <span>
              # comments
            </span>
            <svg stroke="currentColor"
                 fill="currentColor"
                 stroke-width="0"
                 viewbox="0 0 320 512"
                 height="1em"
                 width="1em"
                 xmlns="http://www.w3.org/2000/svg"
            >
              <path d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z">
              </path>
            </svg>
          </button>
        </th>
        <th class="edit-article__actions-container">
          Actions
        </th>
      </tr>
      // TODO: Detach logic of rendering
      <div class="sc-bYUneI sc-gXCJSa dWksIS cuYcqn">
        <div data-testid="three-dots-loading"
             aria-label="three-dots-loading"
             aria-busy="true"
             role="status"
             style="display: flex;"
        >
          <svg width="80"
               height="80"
               viewbox="0 0 120 30"
               xmlns="http://www.w3.org/2000/svg"
               fill="#007BFF"
               data-testid="three-dots-svg"
          >
            <circle cx="15"
                    cy="15"
                    r="15"
            >
              <animate attributename="r"
                       from="15"
                       to="15"
                       begin="0s"
                       dur="0.8s"
                       values="15;9;15"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
              <animate attributename="fill-opacity"
                       from="1"
                       to="1"
                       begin="0s"
                       dur="0.8s"
                       values="1;.5;1"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
            </circle>
            <circle cx="60"
                    cy="15"
                    r="9"
                    attributename="fill-opacity"
                    from="1"
                    to="0.3"
            >
              <animate attributename="r"
                       from="9"
                       to="9"
                       begin="0s"
                       dur="0.8s"
                       values="9;15;9"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
              <animate attributename="fill-opacity"
                       from="0.5"
                       to="0.5"
                       begin="0s"
                       dur="0.8s"
                       values=".5;1;.5"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
            </circle>
            <circle cx="105"
                    cy="15"
                    r="15"
            >
              <animate attributename="r"
                       from="15"
                       to="15"
                       begin="0s"
                       dur="0.8s"
                       values="15;9;15"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
              <animate attributename="fill-opacity"
                       from="1"
                       to="1"
                       begin="0s"
                       dur="0.8s"
                       values="1;.5;1"
                       calcmode="linear"
                       repeatcount="indefinite"
              >
              </animate>
            </circle>
          </svg>
        </div>
      </div>
    </table>
  </form>
</div>
`;
